apps/institution-service/src/controllers/institutionController.ts

```diff
@@ -97,3 +98,84 @@ export const updateInstitution = async (req: Request, res: Response) => {
       .json({ error: "An error occurred while updating the institution" });
   }
 };
+
+interface AggregatorIntegrator {
+  aggregator_institution_id: string;
+  supports_oauth: boolean;
+  supports_identification: boolean;
+  supports_verification: boolean;
+  supports_aggregation: boolean;
+  supports_history: boolean;
+  isActive: boolean;
+  aggregator: {
+    name: string;
+    id: number;
+    displayName: string | null;
+    logo: string | null;
+  };
+}
+
+interface InstitutionDetail {
+  id: string;
+  name: string;
+  keywords: string[];
+  logo: string;
+  url: string;
+  is_test_bank: boolean;
+  routing_numbers: string[];
+  createdAt: string;
+  updatedAt: string;
+  aggregatorIntegrators: AggregatorIntegrator[];
+}
+export interface PaginatedInstitutionsResponse {
+  currentPage: number;
+  pageSize: number;
+  totalRecords: number;
+  totalPages: number;
+  institutions: InstitutionDetail[];
+}
+
+export const getPaginatedInstitutions = async (req: Request, res: Response) => {
+  try {
+    const { limit, offset, page } = res.locals.pagination! as PaginationOptions;
```

Comment:
I feel like not using middleware would be simpler and better.

Looking at this code I had to look up what the heck locals are, and go figure out how we were attaching locals to the response.

I'd prefer something like const { limit, offset, page } = getPaginationOptions(req)

Reply:
It also makes testing more realistic, because our tests will use the actual request object instead of the assuming the locals object is populated correctly

Reply:
That's how I had it originally until I moved it into a middleware, I'll switch it back